internal/common/dev_scripts_aspect.bzl

```diff
@@ -0,0 +1,41 @@
+# Copyright 2019 The Bazel Authors. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+"""Aspect to collect dev scripts from `deps` attribute.
+"""
+
+load("//internal/ng_apf_library:ng_apf_library.bzl", "ScriptsProvider")
```

Comment:
I think it would be better for that provider to be defined outside the layering violation code. ng_apf_library should probably import this symbol from somewhere else.
(since Greg is likely adding another kind of node_module generated rule that runs browserify or something)